src/renderer/qssWidget/js/qssQuickFoldersWidget.js

```diff
@@ -0,0 +1,173 @@
+            searchField: ".flc-search",
+            searchButton: ".flc-searchButton",
+            labelTitle: ".flc-qssSearchWidget-label",
+            errorMessage: ".flc-qssSearchWidget-errorMessage"
+        },
+
+        enableRichText: true,
+
+        model: {
+            setting: {},
+            value: "{that}.model.setting.value",
+            messages: {
+                labelTitle: "{quickFolders}.model.messages.label",
+                searchButtonLabel: "{quickFolders}.model.messages.searchButtonLabel",
+                alertLabel: "{quickFolders}.model.messages.alertLabel"
+            },
+            errorMessageSelector: "{that}.options.selectors.errorMessage",
+        },
+
+        listeners: {
+            onCreate: {
+                func: "{that}.onCreate"
+            }
+        },
+        invokers: {
+            onCreate: {
+                funcName: "gpii.qssWidget.quickFolders.onCreate",
+                args: ["{gpii.qssWidget.quickFolders}.errorMessage"] // sending the errorMessage component
+            }
+        },
+
+        components: {
+            searchField: {
+                type: "gpii.psp.widgets.textfield",
+                container: "{that}.dom.searchField",
+                options: {
+                    model: {
+                        value: "{gpii.qssWidget.quickFolders}.model.value"
+                    },
+                    invokers: {
+                        onSearch: {
+                            funcName: "gpii.qssWidget.quickFolders.onSearch",
+                            args: [
+                                "{that}.model.value", // search field value
+                                "{gpii.qssWidget.quickFolders}.errorMessage", //errorMessage component
+                                "{gpii.qssWidget.quickFolders}.options.siteConfig.morphicQuickFolderPath" // siteConfig's base folder
+                            ]
+                        }
+                    }
+                }
+            },
+            searchButton: {
+                type: "gpii.psp.widgets.button",
+                container: "{that}.dom.searchButton",
+                options: {
+                    model: {
+                        label: "{gpii.qssWidget.quickFolders}.model.messages.searchButtonLabel"
+                    },
+                    invokers: {
+                        onClick: "{gpii.qssWidget.quickFolders}.searchField.onSearch"
+                    }
+                }
+            },
+            errorMessage: {
+                type: "gpii.psp.widgets.alert",
+                container: "{that}.dom.errorMessage",
+                options: {
+                    model: {
+                        label: "{gpii.qssWidget.quickFolders}.model.messages.alertLabel"
+                    },
+                    styles: {
+                        alertHidden: "fl-qssSearchWidget-alert-hidden"
+                    },
+                    invokers: {
+                        onError: {
```

Comment:
onError and onSuccess should be events, with these bodies added as listeners. Namespaces for these impls can be "show" and "hide", the implementation is small enough that can be expressed inline, e.g.
listeners: {
    "onError.hide": {
        "this": "{that}.container",
        method: "removeClass",
        args: "{that}.options.styles.alertHidden"
    }